platforms/core-configuration/declarative-dsl-core/src/testFixtures/kotlin/org/gradle/internal/declarativedsl/dom/mutation/AnalysisSchemaTestUtils.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.declarativedsl.dom.mutation
+
+import org.gradle.declarative.dsl.schema.AnalysisSchema
+import org.gradle.declarative.dsl.schema.DataClass
+import org.gradle.internal.declarativedsl.schemaUtils.findTypeFor
+
+
+inline fun <reified T> AnalysisSchema.typeFor(): DataClass =
```

Comment:
We already have a very similar `typeFor` implementation in `org.gradle.internal.declarativedsl.schemaUtils`. Does that function not work for some reason?

Reply:
The `typeFor` methods were/are only used in tests, so I've moved them accordingly. The duplication you suspect doesn't exist when you take the entirety of the changes into account.

Reply:
Ah, now I see. Then let's keep the original function, see https://github.com/gradle/gradle/pull/30641#discussion_r1774914877.